rfcs/rating-component.md

```diff
@@ -0,0 +1,96 @@
+# Rating Component
+
+The `Rating` component includes two main types:
+
+* `STAR` (default)
+* `EMOTICON`
+
+## Usage
+
+### Basic usage (Star)
+
+```javascript
+import * as React from 'react';
+import {Rating} from 'baseui/rating';
+
+export default () => <Rating />;
+```
+
+### Advanced usage (Emoticon)
+
+```javascript
+import * as React from 'react';
+import {Rating, KIND} from 'baseui/rating';
+
+class EmoticonExample extends React.Component {
+  state = {
+    rating: 1
+  };
+
+  render() {
+    return (
+      <Rating
+        kind={KIND.EMOTICON}
+        value={this.state.rating}
+        onChange={({value}) => {
+          this.setState({ rating: value });
+        }} />
+    );
+  }
+}
+
+export default EmoticonExample;
+```
+
+## Exports
+
+* `Rating`
+* `StatefulContainer`
+* `StyledRoot`
+* `StyledStar`
+* `StyledEmoticon`
+* `STATE_CHANGE_TYPE`
+* `KIND`
+
+## `Rating` API
+
+* `kind: KIND` - Optional
+  The type of `Rating` to display
+* `value: number` - Optional
+  The current rating value
+* `onChange: ({value}) => void` - Optional
+  Callback that returns a newly selected value
+* `overrides: {}` - Optional
+  * `Root: ?React.ComponentType` component to use for root Rating styling
+  * `Star: ?React.ComponentType` component to use for star ratings
+  * `Emoticon: ?React.ComponentType` component to use for emoticon ratings
```

Comment:
if emoticons are override-able, does the index (or explicit enum) need to be provided?

Reply:
I've added 2 new props to the styled-components to address this. `$isSelected` and `$index`